bucketlist/auth/views.py

```diff
@@ -0,0 +1,136 @@
+                    response = {
+                        'message': 'Sorry Invalid email. please put a valid email'
+                    }
+                     # return a response notifying the user that email is invalid
+                    return make_response(jsonify(response)), 400
+
+                check_username = re.match('^[a-zA-Z0-9_.-]+$', username)
+                if check_username is None:
+                    response = {
+                        'message': 'Sorry Invalid Username format. please put a valid username'
+                    }
+                     # return a response notifying the user that credentials username is invalid
+                    return make_response(jsonify(response)), 400
+
+                user = User(username=username, email=email, password=password)
+                user.save()
+
+                response = {
+                    'message': 'You registered successfully. Please log in.'
+                }
+                # return a response notifying the user that they registered successfully
+                return make_response(jsonify(response)), 201
+            except Exception as e:
+                # An error occured, therefore return a string message containing the error
+                response = {
+                    'message': str(e)
+                }
+                return make_response(jsonify(response)), 401
+        else:
+            # There is an existing user. We don't want to register users twice
+            # Return a message to the user telling them that they they already exist
+            response = {
+                'message': 'User already exists. Please login.'
+            }
+
+            return make_response(jsonify(response)), 409
+
+
+class LoginView(MethodView):
+    """This class-based view handles user login and access token generation."""
+
+    def post(self):
+        """Handle POST request for this view."""
+        try:
+            # Get the user object using their email (unique to every user)
+            user = User.query.filter_by(email=request.data['email']).first()
+
+            # Try to authenticate the found user using their password
+            if user and user.password_is_valid(request.data['password']):
+                # Generate the access token. This will be used as the authorization header
+                access_token = user.generate_token(user.id)
+                if access_token:
+                    response = {
+                        'message': 'You logged in successfully.',
+                        'access_token': access_token.decode()
+                    }
+                    return make_response(jsonify(response)), 200
+            else:
+                # User does not exist. Therefore, we return an error message
+                response = {
+                    'message': 'Invalid email or password, Please try again'
+                }
+                return make_response(jsonify(response)), 400
+
```

Comment:
You can try to remove `else` if you use `if not user and not user.` then return the invalid credentials message. Line 97